tests/tests/RDMResponderTest.cpp

```diff
@@ -0,0 +1,386 @@
+  SendRequest(request.get());
+
+  request.reset(NewDiscoveryUniqueBranchRequest(
+      m_controller_uid, UID(m_our_uid.ManufacturerId(), 0),
+      UID::VendorcastAddress(m_our_uid), 0));
+  SendRequest(request.get());
+}
+
+TEST_F(RDMResponderTest, testMute) {
+  RDMResponder_Initialize(TEST_UID, SendResponse);
+
+  const uint8_t expected_data[] = {
+    0xcc, 0x01, 24,
+    0x7a, 0x70, 0x10, 0x00, 0x00, 0x00,  // dst UID
+    0x7a, 0x70, 0x00, 0x00, 0x00, 0x00,  // src UID
+    0x00, 0x00, 0x00, 0x00, 0x00,
+    0x11, 0x00, 0x02, 0x00, 0x02, 0xdc
+  };
+
+  EXPECT_CALL(sender_mock,
+              SendResponse(true, _, _))
+     .With(Args<1, 2>(PayloadIs(expected_data, arraysize(expected_data))))
+     .Times(1);
+
+  EXPECT_FALSE(RDMResponder_IsMuted());
+  unique_ptr<RDMDiscoveryRequest> request(NewMuteRequest(
+      m_controller_uid, m_our_uid, 0));
+  SendRequest(request.get());
+  EXPECT_TRUE(RDMResponder_IsMuted());
+
+  // Broadcasts should return no response
+  request.reset(NewMuteRequest(m_controller_uid, UID::AllDevices(), 0));
+  SendRequest(request.get());
+
+  // Similarly vendorcasts shouldn't trigger a response
+  request.reset(NewMuteRequest(
+      m_controller_uid, UID::VendorcastAddress(m_our_uid), 0));
+  SendRequest(request.get());
+}
+
+TEST_F(RDMResponderTest, testUnMute) {
+  RDMResponder_Initialize(TEST_UID, SendResponse);
+
+  const uint8_t expected_data[] = {
+    0xcc, 0x01, 24,
+    0x7a, 0x70, 0x10, 0x00, 0x00, 0x00,  // dst UID
+    0x7a, 0x70, 0x00, 0x00, 0x00, 0x00,  // src UID
+    0x00, 0x00, 0x00, 0x00, 0x00,
+    0x11, 0x00, 0x03, 0x00, 0x02, 0xdd
+  };
+
+  // Send a broadcast mute first.
+  unique_ptr<RDMDiscoveryRequest> request(NewMuteRequest(
+      m_controller_uid, UID::AllDevices(), 0));
+  SendRequest(request.get());
+  EXPECT_TRUE(RDMResponder_IsMuted());
+
+  EXPECT_CALL(sender_mock,
+              SendResponse(true, _, _))
+     .With(Args<1, 2>(PayloadIs(expected_data, arraysize(expected_data))))
+     .Times(1);
+
+  request.reset(NewUnMuteRequest(m_controller_uid, m_our_uid, 0));
+  SendRequest(request.get());
+  EXPECT_FALSE(RDMResponder_IsMuted());
+
+  // Broadcasts should return no response
+  request.reset(NewUnMuteRequest(m_controller_uid, UID::AllDevices(), 0));
+  SendRequest(request.get());
+
+  request.reset(NewUnMuteRequest(
+      m_controller_uid, UID::VendorcastAddress(m_our_uid), 0));
+  SendRequest(request.get());
+}
+
+
+TEST_F(RDMResponderTest, subdeviceNack) {
+  RDMResponder_Initialize(TEST_UID, SendResponse);
+
+  const uint8_t expected_data[] = {
+    0xcc, 0x01, 0x1a,
+    0x7a, 0x70, 0x10, 0x00, 0x00, 0x00,  // dst UID
+    0x7a, 0x70, 0x00, 0x00, 0x00, 0x00,  // src UID
+    0x00, 0x02, 0x00, 0x00, 0x00,
+    0x21, 0x00, 0x60, 0x2,
+    0x00, 0x09,
+    0x03, 0x59
+  };
+
+  EXPECT_CALL(sender_mock,
+              SendResponse(true, _, _))
+     .With(Args<1, 2>(PayloadIs(expected_data, arraysize(expected_data))))
+     .Times(1);
+
+  // 0x1fff isn't a PID (yet!)
```

Comment:
The test is only correct because 0x1fff isn't a PLASA PID. That's what the comment is supposed to indicate.